ffmpeg.c

```diff
@@ -589,13 +589,15 @@ static int ffmpeg_codec_is_blacklisted(const char *codec_name){
 
     static const char *blacklisted_codec[] =
     {
+#if (!((LIBAVCODEC_VERSION_MAJOR >= 58) && (LIBAVCODEC_VERSION_MINOR >= 55)))
```

Comment:
Once the h264_v4l2m2m is merged, can this be changed to a condition on libavformat?  I believe it makes it easier if we just track a single component version number of ffmpeg in the Motion code.  This is what is being referenced in the comment at the top of the ffmpeg.c module.

Reply:
Might be cleaner to remove the blacklist and move this check into `ffmpeg_set_codec_preferred` function, consistent with `h264_v4l2m2m` and change the condition to
```c
#if (LIBAVFORMAT_VERSION_MAJOR > 58) || ( (LIBAVFORMAT_VERSION_MAJOR == 58) && ( (LIBAVFORMAT_VERSION_MINOR > 29) || ((LIBAVFORMAT_VERSION_MINOR == 29) && (LIBAVFORMAT_VERSION_MICRO > 100)) ) )
    ffmpeg->preferred_codec = USER_CODEC_H264OMX;
#else
    MOTION_LOG(ERR, TYPE_ENCODER, NO_ERRNO
        ,_("%s codec has been blacklisted"), ffmpeg->codec_name);
    ffmpeg_free_context(ffmpeg);
    return -1;
#endif
```

Reply:
Agreed.